persistence/soft-index/src/main/java/org/infinispan/persistence/sifs/SoftIndexFileStore.java

```diff
@@ -190,6 +193,26 @@ public void start() {
       logAppender.setSeqId(maxSeqId.get() + 1);
    }
 
+   private String sanitizedCacheName() {
+      return ctx.getCache().getName().replaceAll("[^a-zA-Z0-9-_\\.]", "_");
+   }
+
+   protected String getQualifiedDataLocation() {
+      if (configuration.dataLocation() == null) {
+         return Paths.get(ctx.getGlobalStateConfiguration().persistentLocation(), sanitizedCacheName(), "data").toString();
+      } else {
+         return Paths.get(configuration.dataLocation(), sanitizedCacheName(), "data").toString();
+      }
+   }
```

Comment:
As these methods are pretty much the same for all stores, we should probably add a generic static method to `PersistenceUtil.java` in core.

Something like:

```java
public static String getQualifiedLocation(Supplier<String> location, GlobalStateConfiguration globalCfg, String cacheName, String dataDir) {
  // sanitize and return location
}
````

Reply:
Done